onboard/onboard.py

```diff
@@ -63,10 +62,34 @@ def create_sg_repo(self, pkg_name: str) -> GitProject:
 
         return project
 
-    def run(self, pkg_name: str, branch: str, skip_build: bool = False):
+    @staticmethod
+    def get_distgit_branch(pkg_name: str) -> Optional[str]:
+        dg_token = getenv("DISTGIT_TOKEN")
+        if not dg_token:
+            logger.error("No DISTGIT_TOKEN specified.")
+            return None
+        service = PagureService(token=dg_token, instance_url="https://git.centos.org/")
+        dg_project = service.get_project(namespace="rpms", repo=pkg_name)
+        branches = dg_project.get_branches()
+        for b in ["c8s", "c8"]:
+            if b in branches:
+                return b
+        else:
+            logger.error("No c8s or c8 branches in dist-git.")
+            return None
+
+    def run(self, pkg_name: str, branch: str = None, skip_build: bool = False):
         action = "Updating" if self.update else "Onboarding"
+
+        if not branch:
+            logger.info("No branch specified. Discovering dist-git.")
```

Comment:
nitpick: I'd say "Inspecting" should be a better word - I understand you "discover" things which are not known to people, we know what dist-git is.

Reply:
changed